### PR TITLE
User site feedback

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: app/assets/stylesheets/frontend/views/feedback.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  .radio_buttons {
+    margin-right: 5px;
+    vertical-align: top;
+  }
+
+  textarea {
+    margin-right: 0;
+    margin-left: 0;
+    -webkit-box-sizing: border-box;
+       -moz-box-sizing: border-box;
```
Comment (matthewford) [inline app/assets/stylesheets/frontend/views/feedback.scss]: Line should be indented 4 spaces, but was indented 7 spaces<br>Avoid vendor prefixes.

File: app/assets/stylesheets/frontend/views/feedback.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            box-sizing: border-box;
```
Comment (matthewford) [inline app/assets/stylesheets/frontend/views/feedback.scss]: Line should be indented 4 spaces, but was indented 12 spaces

File: app/assets/stylesheets/frontend/views/feedback.scss [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  .radio_buttons {
+    margin-right: 5px;
+    vertical-align: top;
+  }
+
+  textarea {
+    margin-right: 0;
+    margin-left: 0;
+    -webkit-box-sizing: border-box;
```
Comment (matthewford) [inline app/assets/stylesheets/frontend/views/feedback.scss]: Properties should be ordered margin-right, margin-left, -moz-box-sizing, -webkit-box-sizing, box-sizing<br>Avoid vendor prefixes.

File: app/assets/stylesheets/frontend/views/feedback.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
+.page-feedback & {
+  span.radio {
```
Comment (matthewford) [inline app/assets/stylesheets/frontend/views/feedback.scss]: Avoid qualifying class selectors with an element.

File: app/assets/stylesheets/frontend/views/feedback.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    display: block;
+  }
+
+  .question-block:first-child h2 {
+    margin-top: 0;
+  }
+
+  .radio_buttons {
```
Comment (matthewford) [inline app/assets/stylesheets/frontend/views/feedback.scss]: Selector `radio_buttons` should be written in lowercase with hyphens

File: app/assets/stylesheets/qae.scss [hunk 6]
```diff
@@ -48,6 +48,7 @@ $gutter-one-sixth: $gutter/6;
   @import 'frontend/views/dashboard';
   @import 'frontend/views/eligibility';
   @import 'frontend/views/award_form';
+  @import 'frontend/views/feedback';
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Prefer double-quoted strings

File: app/models/site_feedback.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class SiteFeedback < ActiveRecord::Base
+  extend Enumerize
+
+  enumerize :rating, in: { very_satisfied: 5, satisfied: 4, neither_satisfied_or_dissatisfied: 3, dissatisfied: 2, very_dissatisfied: 1 }
```
Comment (matthewford) [inline app/models/site_feedback.rb]: Line is too long. [137/100]

File: app/assets/stylesheets/qae.scss [hunk 8]
```diff
@@ -1,57 +1,58 @@
...
-@import '_grid_layout';
-@import '_conditionals';
-@import '_colours';
-@import '_css3';
-@import '_typography';
-@import '_device-pixels';
-@import '_measurements';
-@import 'design-patterns/_buttons';
-@import 'design-patterns/_alpha-beta';
+@import "_shims";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_shims` should be written as `shims`

File: app/assets/stylesheets/qae.scss [hunk 9]
```diff
@@ -1,57 +1,58 @@
...
+@import "_grid_layout";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_grid_layout` should be written as `grid_layout`

File: app/assets/stylesheets/qae.scss [hunk 10]
```diff
@@ -1,57 +1,58 @@
...
+@import "_conditionals";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_conditionals` should be written as `conditionals`

File: app/assets/stylesheets/qae.scss [hunk 11]
```diff
@@ -1,57 +1,58 @@
...
+@import "_colours";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_colours` should be written as `colours`

File: app/assets/stylesheets/qae.scss [hunk 12]
```diff
@@ -1,57 +1,58 @@
...
+@import "_css3";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_css3` should be written as `css3`

File: app/assets/stylesheets/qae.scss [hunk 13]
```diff
@@ -1,57 +1,58 @@
...
+@import "_typography";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_typography` should be written as `typography`

File: app/assets/stylesheets/qae.scss [hunk 14]
```diff
@@ -1,57 +1,58 @@
...
+@import "_device-pixels";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_device-pixels` should be written as `device-pixels`

File: app/assets/stylesheets/qae.scss [hunk 15]
```diff
@@ -1,57 +1,58 @@
...
+@import "_measurements";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_measurements` should be written as `measurements`

File: app/assets/stylesheets/qae.scss [hunk 16]
```diff
@@ -1,57 +1,58 @@
...
+@import "design-patterns/_buttons";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `design-patterns/_buttons` should be written as `design-patterns/buttons`

File: app/assets/stylesheets/qae.scss [hunk 17]
```diff
@@ -1,57 +1,58 @@
...
+@import "design-patterns/_alpha-beta";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `design-patterns/_alpha-beta` should be written as `design-patterns/alpha-beta`

File: app/assets/stylesheets/qae.scss [hunk 18]
```diff
@@ -1,57 +1,58 @@
...
-@import 'frontend/forms';
-@import 'frontend/attachment';
+@import "frontend/smart_answers";
+@import "frontend/layout";
+@import "frontend/forms";
+@import "frontend/attachment";
 
 //
-@import '_qae-slimmer-patch';
+@import "_qae-slimmer-patch";
```
Comment (matthewford) [inline app/assets/stylesheets/qae.scss]: Imported partial `_qae-slimmer-patch` should be written as `qae-slimmer-patch`

**Comment by RuslanHamidullin** [comment]
All Good :+1: 

2 notes (probably for future):
1) Need to add validation on comment text by max length
2) Need captcha (otherwise would be a lot of span)

**Comment by matthewford** [comment]
do we need a captcha? i think lets hold off on that till it becomes an issue we have cloudflare

**Comment by RuslanHamidullin** [comment]
Right ^. @dreamfall I forget - we have Cloudfare, which should care about robots.
Probably captcha wouldn't need.

**Pull request merged**